### PR TITLE
Add a diesel and diesel-async example

### PR DESCRIPTION
**Pull request by weiznich**
## Motivation

Diesel is a popular ORM that is often used in combination with web frameworks like axum. I've noticed that there are no examples for this in the example directory yet. 

## Solution

Add two examples, one based on the main (sync) diesel crate and one based on the slightly experimental diesel-async version.

### CODE REVIEW

**Comment by weiznich** [comment]
This addresses https://github.com/tokio-rs/axum/issues/475. Any change to get a review for this?

File: examples/diesel-async-postgres/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+type Pool = bb8::Pool<AsyncDieselConnectionManager<AsyncPgConnection>>;
+
+#[tokio::main]
+async fn main() {
+    tracing_subscriber::registry()
+        .with(
+            tracing_subscriber::EnvFilter::try_from_default_env()
+                .unwrap_or_else(|_| "example_tokio_postgres=debug".into()),
```
Comment (davidpdrsn) [inline examples/diesel-async-postgres/src/main.rs:60]:
```suggestion
                .unwrap_or_else(|_| "example_diesel_async_postgres=debug".into()),
```

File: examples/diesel-postgres/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                .returning(User::as_returning())
+                .get_result(conn)
+        })
+        .await
+        .map_err(internal_error)?
+        .map_err(internal_error)?;
+    Ok(Json(res))
+}
+
+#[axum_macros::debug_handler]
```
Comment (davidpdrsn) [inline examples/diesel-postgres/src/main.rs:99]:
```suggestion
```

File: examples/diesel-postgres/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    // run it with hyper
+    let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
+    tracing::debug!("listening on {}", addr);
+    axum::Server::bind(&addr)
+        .serve(app.into_make_service())
+        .await
+        .unwrap();
+}
+
+#[axum_macros::debug_handler]
```
Comment (davidpdrsn) [inline examples/diesel-postgres/src/main.rs:80]:
```suggestion
```

File: examples/diesel-postgres/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,119 @@
+//! Run with
+//!
+//! ```not_rust
+//! cargo run -p example-diesel-postgres
```
Comment (davidpdrsn) [inline examples/diesel-postgres/src/main.rs:4]: Do we need a command here to run migrations?

File: examples/diesel-async-postgres/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    // run it with hyper
+    let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
+    tracing::debug!("listening on {}", addr);
+    axum::Server::bind(&addr)
+        .serve(app.into_make_service())
+        .await
+        .unwrap();
+}
+
+#[axum_macros::debug_handler]
```
Comment (weiznich) [inline examples/diesel-async-postgres/src/main.rs:86]:
```suggestion
```

File: examples/diesel-postgres/src/main.rs (hunk 4 above)
Comment (weiznich) [reply examples/diesel-postgres/src/main.rs:4]: Yes that was missing.
Rather then requiring a separate step there I just used `diesel_migration` to automatically run the migrations on application startup. 
See https://github.com/tokio-rs/axum/pull/1853/commits/0abd85cf9e79eb4fa0066c0f0ce2880d71042817

(That's not possible with diesel-async yet, so no change there)

**Review by davidpdrsn — APPROVED** [review]
Thanks!

**Comment by davidpdrsn** [comment]
Alright thanks! It used to work though but I'll give that a shot.

**Comment by davidpdrsn** [comment]
`brew install libpq` didn't fix it 😞

**Comment by weiznich** [comment]
Hmm, that's interesting. The targets indicates you use an arm m1/m2 mac? In that case I would start with checking whether the arm version of libpq is installed or the x86 version. If it's the arm version you can try setting `PQ_LIB_DIR` so that it points to the directory that contains `libpq.so`. That should normally fix the issue, although I'm a bit surprised that it doesn't pick up the correct lib by default (that normally works on linux/mac os)

**Pull request merged**